### ISSUE TITLE
Tests run in test explorer never ends 

### ISSUE THREAD

**Issue by vitaliipanchenko** [fscheck/FsCheck#569]
Steps to reproduce:
1 Create F# XUnit project template 
2 run following test in test explorer 

```
open FsCheck.Xunit

type Member () =
    member val Name = "" with get, set

[<Property>]
let ``test mutation`` (m :Member)  =
    m.Name <- "1"
    m.Name = "0"
```

OS Windows 10
IDE VS 16.9.4, Rider 2021.1.2
FsCheck.Xunit 2.15.2

**Comment by ploeh** [comment]
FWIW, I can reproduce in both Visual Studio Community 2019 16.9.4 and `dotnet test`.

**Comment by kurtschelfthout** [comment]
Yes, this is because `m` is captured for shrinking. So when the test fails, FsCheck tries to shrink `m` with `Name` set to "1". It tries to shrink to say "", the empty string. But to check if the shrink is succesful, i.e. if the test is still failing, it needs to run the test. Which then sets `Name` to "1" again, the test fails again, and FsCheck tries to shrink the "1" to "" again, ad infinitum.

Basically if you want to mutate an object inside a test, you'll need to copy it first.

**Comment by vitaliipanchenko** [comment]
As proposal maybe it is worth to add custom exception for the case

**Comment by kurtschelfthout** [comment]
I don't really see how to detect this case in particular (at least not without adding a bunch of complexity). 

I think the best thing to do is to limit the number of shrink steps to some maximum, so at least it doesn't hang. Can also add some explanation of this possible cause in case that limit is hit.

**Comment by kurtschelfthout** [comment]
Released in 2.15.3
